pkg/app/app.go

```diff
@@ -91,6 +101,61 @@ func defaultFuncApiUrl(logger *zerolog.Logger) configuration.DefaultValueFunctio
 	return callback
 }
 
+// oauthApiUrl returns the API URL specified by the audience claim in a JWT
+// token established by a prior OAuth authentication flow.
+//
+// Returns an empty string if an OAuth token is not available, cannot be parsed,
+// or lacks such an audience claim, along with an error that may have occurred
+// in the attempt to parse it.
+func oauthApiUrl(config configuration.Configuration) (string, error) {
+	oauthTokenString, ok := config.Get(auth.CONFIG_KEY_OAUTH_TOKEN).(string)
+	if !ok || oauthTokenString == "" {
+		return "", nil
+	}
+	var token oauth2.Token
+	if err := json.Unmarshal([]byte(oauthTokenString), &token); err != nil {
+		return "", err
+	}
+	return readAudience(&token)
+}
+
+// readAudience returns the first audience claim from an OAuth2 access token, or
+// an error which prevented its parsing.
+//
+// If the claim is not present, an empty string is returned.
+//
+// This function was derived from https://pkg.go.dev/golang.org/x/oauth2/jws#Decode,
+// which is licensed as follows:
+//
+// Copyright 2014 The Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+func readAudience(token *oauth2.Token) (string, error) {
+	// decode returned id token to get expiry
+	s := strings.Split(token.AccessToken, ".")
+	if len(s) < 2 {
+		// TODO(jbd): Provide more context about the error.
+		return "", errors.New("jws: invalid token received")
+	}
+	decoded, err := base64.RawURLEncoding.DecodeString(s[1])
+	if err != nil {
+		return "", err
+	}
+	c := struct {
+		// NOTE: The original jws package models audience with a string, not a
+		// []string. This fails to parse Snyk JWTs.
+		Aud []string `json:"aud"`
```

Comment:
it doesn't feel correct to use the `aud` claim for fetching tokens.
We are using aud claim to determine where to fetch the access token from. 
Afaik aud claim should only be used to identify the audience/recipients for this token.
https://datatracker.ietf.org/doc/html/rfc7519#section-4.1.3